all_test.go

```diff
@@ -159,6 +163,59 @@ func hasValidHeader(path string, r io.Reader) (bool, error) {
 	return headerRegex.Match(allBytes), nil
 }
 
+func TestGolangVersion(t *testing.T) {
+	goVersions := make(map[string][]string)
+	sfs := os.DirFS(".")
+	err := fs.WalkDir(sfs, ".", func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if strings.HasSuffix(path, "Dockerfile") {
```

Comment:
Any reason for using a suffix here and below, rather than equality using `d.Name()`?

Basically I guess I'm asking whether we *want* to check files of "SittingOnTheDockerfile" and "letmego.mod"

Reply:
My assumption is we may have multiple Dockerfiles (we have 2 within `infra`) and the file name maybe `something.Dockerfile` or `Dockerfile`. In either case, the test can check the go version.

The same with `go.mod`.